spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -596,7 +596,7 @@ private static String sqlString(org.apache.iceberg.expressions.Literal<?> lit) {
       if (lit.value() instanceof String) {
         return "'" + lit.value() + "'";
       } else if (lit.value() instanceof ByteBuffer) {
-        throw new IllegalArgumentException("Cannot convert bytes to SQL literal: " + lit);
+        return lit.toString();
```

Comment:
Would these need to be quoted as SQL literals? I believe, at least with Spark, that there is a leading x followed by single quotes around the rest.

Eg. when I open a spark-shell, I use the following to represent bytes.

```
spark.sql("select x'0FA67B' as value").show
+----------+
|     value|
+----------+
|[0F A6 7B]|
+----------+

scala> spark.sql("select x'0FA67B' as value").printSchema
root
 |-- value: binary (nullable = false)
```

Reply:
The problem now is that as long as the query filter column is of the binary type, no matter what the filter condition is, it will report an error directly.

Reply:
I think @kbendick is right. This is supposed to produce a literal value that Spark can parse. If you're going to implement it, then it needs to produce the correct format. Looks like that is `x'<hex>'`.